### PR TITLE
Ncon tests

### PR DESCRIPTION
**Pull request by amilsted**
Improve `ncon()` error handling and add some tests.

Not yet tested: `ncon_network()`. Although it is implicitly tested via `ncon()`, the interface should probably be subject to tests.

### CODE REVIEW

File: tensornetwork/ncon_interface.py [hunk 1]
```diff
@@ -147,6 +176,12 @@ def _build_network(tensors, network):
     nodes = []
     edges = {}
     for (i, (tensor, edge_lbls)) in enumerate(zip(tensors, network)):
+        # Allow single indices to be passed as scalars.
+        try:
+            edge_lbls = list(edge_lbls)
```
Comment (Thenerdstation) [inline tensornetwork/ncon_interface.py]: This is gonna cause some issues when edge_lbls are strings.

Comment (amilsted) [reply tensornetwork/ncon_interface.py]: Ah, good point.

**Comment by amilsted** [comment]
Typing is in.

**Comment by Thenerdstation** [comment]
Thanks for changing the spacing to 2 as well!

**Pull request merged**